File: src/test/java/com/uapp_llc/test/LazyInitBeanFactoryPostProcessor.java
package com.uapp_llc.test;

import org.springframework.beans.BeansException;
import org.springframework.beans.factory.config.BeanDefinition;
import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;

public class LazyInitBeanFactoryPostProcessor implements BeanFactoryPostProcessor {

  @Override
  public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
      throws BeansException {
    for (String beanName : beanFactory.getBeanDefinitionNames()) {
      BeanDefinition beanDefinition = beanFactory.getBeanDefinition(beanName);
      beanDefinition.setLazyInit(true);
    }
  }

}

File: src/test/kotlin/com/uapp_llc/test/LazyInitBeanFactoryPostProcessor.kt
package com.uapp_llc.test

import org.springframework.beans.factory.config.BeanFactoryPostProcessor
import org.springframework.beans.factory.config.ConfigurableListableBeanFactory

class LazyInitBeanFactoryPostProcessor : BeanFactoryPostProcessor {

  override fun postProcessBeanFactory(beanFactory: ConfigurableListableBeanFactory) {
    for (beanName in beanFactory.beanDefinitionNames) {
      beanFactory.getBeanDefinition(beanName).isLazyInit = true
    }
  }

}
